### ISSUE TITLE
Local state query by fragment does not work (reactive variable)

### ISSUE THREAD

**Issue by Traktormaster** [apollographql/apollo-client#6877]
**Intended outcome:**
I'd like to have a local-state with a reactive variable queried while expressing that part of the query in a fragment.

I'm not sure if it is supposed to work, but I have not read anything to say otherwise.

**Actual outcome:**
The graphql query is not even sent to the server. The client fails to  create it with "Unknown Fragment for Spread ...".

**How to reproduce the issue:**
I have made a reproduction demo at https://github.com/Traktormaster/react-apollo-error-template/tree/bug/makevar-frag
By default the query will fail:

```js
const ALL_PEOPLE = gql`
  query AllPeople {
    people {
      id
      name
    }

    # TEST-THIS: Using the reactive variable directly is fine
    #connection @client

    # TEST-THIS: Using the reactive variable from a fragment fails
    ...ClientConnection
  }
  ${CONNECTION_FRAGMENT}
`;
```

Swapping around the direct/fragment inclusion of the variable makes it work:
```js
const ALL_PEOPLE = gql`
  query AllPeople {
    people {
      id
      name
    }

    # TEST-THIS: Using the reactive variable directly is fine
    connection @client

    # TEST-THIS: Using the reactive variable from a fragment fails
    #...ClientConnection
  }
  #${CONNECTION_FRAGMENT}
`;
```
 It will also work if I replace the body of the fragment to something else that does not include the reactive variable:
```js
const CONNECTION_FRAGMENT = gql`
  fragment ClientConnection on Query {
    people {
      id
    }
  }
`;
```

**Versions**
apollo-client 3.1.0+ (3.1.3 for sure) has this issue

**Comment by benjamn** [comment]
@Traktormaster @klaussner Ok, I retract my theory! Thanks for the correction, and I'm glad to know that interpolating `DocumentNode`s works as expected. I'll dig into the reproduction.

**Comment by Aleksion** [comment]
Any updates on this?

**Comment by jerelmiller** [comment]
Hey all 👋 

It appears this issue was fixed in [`v3.7.9`](https://github.com/apollographql/apollo-client/releases/tag/v3.7.9) by https://github.com/apollographql/apollo-client/pull/10560. Using the reproduction in the original issue, I was able to successfully get a result as expected where versions prior to 3.7.9 reproduced the bug.

I'm going to go ahead and closed this issue as fixed by #10560. If you see anything else, please feel free to open a new issue. Thanks!
